### DISCUSSION TITLE
Electron renderer socket.io-client receive event with huge delay

### DISCUSSION THREAD

**Discussion by gwendalF** [socketio/socket.io#4238] [Q&A] [answered]
Hello,

I have troubles using the socket.io-client package to my project. I have a python backend server using python-socketio v5 and I try to communicate with it inside my electron app in a react renderer.

I can connect throuh websocket to my server. But I received the custom event emitted with huge delay (~10seconds sometimes more sometimes less) but I can emit to the server immediatly. My server is serve in localhost so I expect to have really small latency. I try to use the package on the main process and it seems to works correctly (receive event directly). 
What can I do to be able to use it on the renderer without issue?

Thanks in adance

Edit: After some digging, even in the main process I have the delay. So the issue is on server side and not with socket.io-client

**Comment by darrachequesne** [comment, marked as answer]
That sounds weird indeed. In localhost, you should have a latency < 10 ms.

Can you add logs on the server side, to check the timings? Or log network events with [netLog](https://www.electronjs.org/docs/latest/api/net-log)?

**Reply by gwendalF** [reply]
It was definetly a serverside issue. The serveur emitted the log of the event emitted but due to an async error the event were not properly emitted

Thanks

**Reply by darrachequesne** [reply]
Oh, OK! Thanks for the update :+1:
